File: 30daysOFcode_Java/day 04.java
import java.io.*;
import java.util.*;
//The Code is crisp and to the point...
/*
note: the solution of this program on hackerrank is
provided wrong, by the problem setter. As, the test solution
adds an extra line at the end of the program but it must not...
*/
public class Person {
    private int age;    
     public Person(int initialAge) {
          age=(initialAge>=0)?initialAge:0;
          if(age==0) System.out.println("Age is not valid, setting age to 0.");
    }
    public void amIOld() {
        //use of a String object saves us from giving
        //multiple print commands...
        String s="";
        if(age<13) s="young.";
        else if(age>12 && age<18) s="teenager.";
        else s="old";
        System.out.println("You are "+s);
    }
    public void yearPasses() {
         age++;
    }
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int T = sc.nextInt();
		for (int i = 0; i < T; i++) {
			int age = sc.nextInt();
			Person p = new Person(age);
			p.amIOld();
			for (int j = 0; j < 3; j++) {
				p.yearPasses();
			}
			p.amIOld();
			System.out.println();
        }
		sc.close();
    }
}

File: 30daysOFcode_Ruby/day 04.rb
class Person
    attr_accessor :age
    def initialize(initialAge)
        if initialAge>0
            @age=initialAge
        else
            @age=0
            puts "Age is not valid, setting age to 0."
        end
    end
    def amIOld()
      # Do some computations in here and print out the correct statement to the console
      if @age<13
            puts "You are young."
      elsif 13<=@age && @age<18
            puts "You are a teenager."
      else
            puts "You are old."
      end
    end
    def yearPasses()
      # Increment the age of the person in here
      @age+=1
    end
end